pkg/skaffold/build/jib/sync.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jib
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"os/exec"
+	"regexp"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+)
+
+type SyncMap map[string]SyncEntry
+
+type SyncEntry struct {
+	Dest     []string
+	FileTime time.Time
+	IsDirect bool
+}
+
+type JSONSyncMap struct {
+	Direct    []JSONSyncEntry `json:"direct"`
+	Generated []JSONSyncEntry `json:"generated"`
+}
+
+type JSONSyncEntry struct {
+	Src  string `json:"src"`
+	Dest string `json:"dest"`
+}
+
+func getSyncMapFromSystem(cmd *exec.Cmd) (*SyncMap, error) {
+	jsm := JSONSyncMap{}
+	stdout, err := util.RunCmdOut(cmd)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get Jib sync map")
+	}
+
+	// To parse the output, search for "BEGIN JIB JSON", then unmarshal the next line into the pathMap struct.
+	// Syncmap is transitioning to "BEGIN JIB JSON: SYNCMAP/1" starting in jib 2.0.0
+	// perhaps this feature should only be included from 2.0.0 onwards? And we generally avoid this?
+	matches := regexp.MustCompile(`BEGIN JIB JSON(?:: SYNCMAP/1)?\r?\n({.*})`).FindSubmatch(stdout)
+	if len(matches) == 0 {
+		return nil, errors.New("failed to get Jib Sync data")
+	}
+
+	line := bytes.Replace(matches[1], []byte(`\`), []byte(`\\`), -1)
```

Comment:
What is this for? Windows support? Also don't see it covered in testcases unless I'm missing something.

Reply:
Good questions, this is copied from another jib-json section, I'll verify what it's doing.

Reply:
https://github.com/GoogleContainerTools/skaffold/pull/2276#discussion_r304542768

Reply:
@chanseokoh great minds...

It's not going to be easy to cover this with integration tests as we don't have Windows integration tests just yet. We _could_ (and should:)) take the output of jib on a Windows env and paste it in our testcases, similarly how we put unix outputs in there.

Reply:
yeah working on that, might need to restructure a few things to actually allow it.

Reply:
I dug into this some more. I think it's an artifact from previous jib outputs, it just happened to work because multiple back slashes are ignored. I think the reality is that we don't need to do this, I have a new commit that removes it, will push tomorrow.

Reply:
done.